app/src/main/java/io/apicurio/registry/services/auth/CustomAuthenticationMechanism.java

```diff
@@ -236,10 +246,38 @@ private Uni<SecurityIdentity> authenticateWithClientCredentials(Pair<String, Str
         }
 
         oidcAuth = new OidcAuth(httpClient, clientCredentials.getLeft(), clientCredentials.getRight());
+        String jwtToken;
+        String credentialsHash = getCredentialsHash(clientCredentials.getLeft() + clientCredentials.getRight());
+        if (cachedAccessTokens.containsKey(credentialsHash) && !cachedAccessTokens.get(credentialsHash).isExpired()) {
+            jwtToken = cachedAccessTokens.get(credentialsHash).getValue();
+        } else {
+            jwtToken = oidcAuth.authenticate();//If we manage to get a token from basic credentials, try to authenticate it using the fetched token using the identity provider manager
+            cachedAccessTokens.put(credentialsHash, new WrappedValue<>(Duration.ofMinutes(5), Instant.now(), jwtToken));
+        }
 
-        final String jwtToken = oidcAuth.authenticate();//If we manage to get a token from basic credentials, try to authenticate it using the fetched token using the identity provider manager
-        oidcAuth.close();
         context.request().headers().set("Authorization", "Bearer " + jwtToken);
         return oidcAuthenticationMechanism.authenticate(context, identityProviderManager);
     }
+
+    private String getCredentialsHash(String credentials) {
+        MessageDigest digest;
+        try {
+            digest = MessageDigest.getInstance("SHA-256");
+            return bytesToHex(digest.digest(credentials.getBytes(StandardCharsets.UTF_8)));
+        } catch (NoSuchAlgorithmException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static String bytesToHex(byte[] hash) {
+        StringBuilder hexString = new StringBuilder(2 * hash.length);
+        for (byte b : hash) {
+            String hex = Integer.toHexString(0xff & b);
+            if (hex.length() == 1) {
+                hexString.append('0');
+            }
+            hexString.append(hex);
+        }
+        return hexString.toString();
+    }
```

Comment:
Can probably replace all this with:

https://github.com/Apicurio/apicurio-registry/blob/master/app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java#L612-L613